paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlActionUtils.java

```diff
@@ -202,6 +202,9 @@ static MySqlSource<String> buildMySqlSource(Configuration mySqlConfig) {
         mySqlConfig
                 .getOptional(MySqlSourceOptions.HEARTBEAT_INTERVAL)
                 .ifPresent(sourceBuilder::heartbeatInterval);
+        mySqlConfig
+                .getOptional(MySqlSourceOptions.SCAN_NEWLY_ADDED_TABLE_ENABLED)
+                .ifPresent(sourceBuilder::scanNewlyAddedTableEnabled);
```

Comment:
Only set this when `MySqlDatabaseSyncMode` is DYNAMIC

Reply:
> Only set this when `MySqlDatabaseSyncMode` is DYNAMIC

Done by setting the option in mysql config when MySqlDatabaseSyncMode is DYNAMIC